app/src/common/shared/com/igalia/wolvic/VRBrowserActivity.java

```diff
@@ -733,14 +756,41 @@ public void onConfigurationChanged(Configuration newConfig) {
 
     @Override
     public void onSharedPreferenceChanged(SharedPreferences sharedPreferences, String key) {
-            if (key.equals(getString(R.string.settings_key_voice_search_service))) {
-                initializeSpeechRecognizer();
-            } else if (key.equals(getString(R.string.settings_key_head_lock))) {
-                boolean isHeadLockEnabled = SettingsStore.getInstance(this).isHeadLockEnabled();
-                setHeadLockEnabled(isHeadLockEnabled);
-                if (!isHeadLockEnabled)
-                    recenterUIYaw(WidgetManagerDelegate.YAW_TARGET_ALL);
+        if (Objects.equals(key, getString(R.string.settings_key_voice_search_service))) {
+            initializeSpeechRecognizer();
+        } else if (Objects.equals(key, getString(R.string.settings_key_head_lock))) {
+            boolean isHeadLockEnabled = mSettings.isHeadLockEnabled();
+            setHeadLockEnabled(isHeadLockEnabled);
+            if (!isHeadLockEnabled)
+                recenterUIYaw(WidgetManagerDelegate.YAW_TARGET_ALL);
+        } else if (Objects.equals(key, getString(R.string.settings_key_tabs_location))) {
+
+            Log.e(LOGTAG, "update tabs location");
+
+            // remove the previous widget
+            if (mTabsBar != null) {
+                Log.e(LOGTAG, "remove previous widget");
+                removeWidget(mTabsBar);
+                mTabsBar.releaseWidget();
+            }
+
+            if (mSettings.getTabsLocation() == SettingsStore.TABS_LOCATION_TRAY) {
```

Comment:
I think a switch-case statement fits better for this use case

Reply:
Done.